buildchain/buildchain/targets/local_image.py

```diff
@@ -212,3 +159,19 @@ def _do_save(self) -> List[types.Action]:
         cmd.append('docker-daemon:{}'.format(self.tag))
         cmd.append('dir:{}'.format(str(self.dirname)))
         return [self.mkdirs, cmd]
+
+    @staticmethod
+    def _expand_dep(dep: Path) -> List[Path]:
+        """Expand a Dockerfile dependency path to regular files."""
+        # Simple file
+        if dep.is_file():
+            return [dep]
+        # Directory or `.`
+        if dep.is_dir():
+            return list(coreutils.ls_files_rec(dep))
+        # Globs - `*` or specific e.g. `*.py`, `*.repo`
+        return [
+            sub_path
+            for sub_path in dep.parent.glob(dep.name)
+            if sub_path.is_file()
+        ]
```

Comment:
What happens if I have the following:
```
COPY src/* dest/
```
And in `src`, I have multiple level of directories, e.g.:
```
src
├── foo
│   └── bar
│       └── baz
└── banana
```
We don't add `src/foo/bar/baz` as a dependency, no ?

Reply:
I didn't change the behaviour for parsing such glob patterns, so I don't really know, and didn't test. I would assume `pathlib.Path.glob` works roughly the same as Docker does when parsing those statements. If it doesn't, I'd suggest we open a tracking issue and tackle it in another PR :wink:

Reply:
Just checked, and indeed, there is a difference between the two. I guess this can be easily improved, by doing something like:

```python
deps = []
for sub_path in dep.parent.glob(dep.name):
    if sub_path.is_file():
        deps.append(sub_path)
    elif sub_path.is_dir():
        deps.extend(coreutils.ls_files_rec(sub_path))
    else:
        # do something? or not?
        continue
return deps

Reply:
See #3359